### PR TITLE
EE-659 File re-encrypt command tool

### PR DESCRIPTION
**Pull request by jorizci**
- Added Aes Ctr algorithm with OpenSSL password/IV flavour
- Refactored encryption core tools
- Added new application: File re-encrypt command tool

### CODE REVIEW

File: cmd-encryption/src/main/java/uk/ac/ebi/ega/cmd/encryption/services/FileReEncryptService.java [hunk 1]
```diff
@@ -59,8 +61,13 @@ public void reEncryptFile(String fireFilePath, String outputPath, OutputFormat o
                     InputStream fireFile = file.getStream();
                     OutputStream outputFile = new FileOutputStream(fileOutputPath)
             ) {
-                ReEncryption.loggedReEncrypt(fireFile, password, decryptionAlgorithm, outputFile, outputPassword,
-                        encryptionAlgorithm);
+                final ReEncryptionReport reEncryptionReport = ReEncryption.loggedReEncrypt(fireFile, password,
+                        decryptionAlgorithm, outputFile, outputPassword, encryptionAlgorithm);
+                if (!Objects.equals(file.getMd5(), reEncryptionReport.getEncryptedMd5())) {
```
Comment (kandaj) [inline cmd-encryption/src/main/java/uk/ac/ebi/ega/cmd/encryption/services/FileReEncryptService.java]: @jorizci  just want to check. if the file encrypted md5s doesn't match will the output file still exists or will be deleted?

Comment (jorizci) [reply cmd-encryption/src/main/java/uk/ac/ebi/ega/cmd/encryption/services/FileReEncryptService.java]: In the command line It won't be deleted (It does on the re-encryption process), we can adjust that and delete the output file on io error/md5 discrepancy, what's is your preference?

Comment (kandaj) [reply cmd-encryption/src/main/java/uk/ac/ebi/ega/cmd/encryption/services/FileReEncryptService.java]: i think its better to delete the file

File: cmd-encryption/src/main/java/uk/ac/ebi/ega/cmd/encryption/options/CmdEncryptOptions.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    public CmdEncryptOptions(String fireFilePath, String outputPath, OutputFormat outputFormat, boolean useFireMount,
+                             int retries, String passwordFile, String outputPasswordFile) {
+        this.fireFilePath = fireFilePath;
+        this.outputPath = outputPath;
+        this.outputFormat = outputFormat;
+        this.useFireMount = useFireMount;
+        this.retries = retries;
+        this.passwordFile = passwordFile;
+        this.outputPasswordFile = passwordFile;
```
[the remark this answers is not in the document]
Comment (jorizci) [reply cmd-encryption/src/main/java/uk/ac/ebi/ega/cmd/encryption/options/CmdEncryptOptions.java]: d'ouch

File: cmd-encryption/pom.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>cmd-encryption</artifactId>
```
[the remark this answers is not in the document]
Comment (jorizci) [reply cmd-encryption/pom.xml]: I'll leave it up to @kandaj

File: cmd-encryption/src/main/java/uk/ac/ebi/ega/cmd/encryption/options/OutputFormat.java [hunk 4]
```diff
@@ -15,21 +15,32 @@
...
-    public static char[] generateRandomString(int size) {
-        char[] randomString = new char[size];
-        for (int i = 0; i < size; i++) {
-            int randomNum = ThreadLocalRandom.current().nextInt(0, dictionary.length());
-            randomString[i] = dictionary.charAt(randomNum);
+    AES_CTR_256_EGA("AES_CTR_256_EGA"),
+
+    AES_CBC_OPENSSL("AES_CBC_OPENSSL");
+
+    private String value;
```
[the remark this answers is not in the document]
Comment (jorizci) [reply cmd-encryption/src/main/java/uk/ac/ebi/ega/cmd/encryption/options/OutputFormat.java]: Needed, discussed offline.

**Pull request merged**